api/types/saml_idp_service_provider.go

```diff
@@ -161,6 +185,26 @@ func (s *SAMLIdPServiceProviderV1) CheckAndSetDefaults() error {
 		s.Spec.EntityID = ed.EntityID
 	}
 
+	if len(s.GetAttributeMapping()) > 0 {
```

Comment:
This check is really redundant here. Just go straight to the "for" loop, and it will be a no-op if the mapping is empty.